aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1852,6 +1811,89 @@ static Tensor& linalg_norm_out_impl(Tensor& result, const Tensor& self, optional
   return result;
 }
 
+static Tensor& linalg_vector_norm_impl(Tensor& result, const Tensor& self, optional<Scalar> opt_ord, optional<IntArrayRef> opt_dim, bool keepdim, optional<ScalarType> opt_dtype) {
+  auto ord = opt_ord.value_or(2).toDouble();
+
+  TORCH_CHECK(self.device().type() == DeviceType::CPU || self.device().type() == DeviceType::CUDA,
+              "norm only supports CPU AND CUDA device type, got: ", self.device().type());
+  TORCH_CHECK(self.layout() == Layout::Strided,
+              "norm only supports strided layout, got: ", self.layout());
+
+  ScalarType in_dtype = opt_dtype.has_value() ? opt_dtype.value() : self.scalar_type();
+  TORCH_CHECK(
+      at::isFloatingType(in_dtype) || at::isComplexType(in_dtype),
+      "Can only calculate the norm of floating point and complex dtypes. Got ",
+      toString(in_dtype),
+      " instead.");
+
+  IntArrayRef dim = opt_dim.value_or(IntArrayRef{});
+
+  // For NumPy compatibility, we can only perform order +/-infinity reduction
+  // (max/min) on a tensor with zero elements if the dimensions to reduce are
+  // nonzero. Otherwise, throw an error.
+  if (self.numel() == 0 && std::abs(ord) == INFINITY) {
+    bool has_identity = true;
+
+    if (dim.size() == 0) {
+      has_identity = false;
+    } else {
+      for (int64_t dim_num : dim) {
+        if (self.size(dim_num) == 0) {
+          has_identity = false;
+          break;
+        }
+      }
+    }
+    if (!has_identity) {
+      TORCH_CHECK(ord != INFINITY, (
+        "cannot perform reduction function max on tensor with "
```

Comment:
"vector_norm cannot..."

The reference to the "reduction function max" is inconsistent with the user's input, which is ord=inf. Maybe:

"vector_norm cannot compute the infinity norm on an empty tensor because the operation does not have an identity"?

Reply:
Sounds good to me